File: src/main/golang/leetcode/editor/cn/[12]Integer to Roman.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func intToRoman(num int) string {
	var (
		thousands = []string{"", "M", "MM", "MMM"}
		hundreds  = []string{"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"}
		tens      = []string{"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"}
		ones      = []string{"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"}
	)
	return thousands[num/1000] + hundreds[(num%1000)/100] + tens[(num%100)/10] + ones[num%10]
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[12]Integer to Roman.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def __init__(self):
        self.search = [
            (1000, "M"),
            (900, "CM"),
            (500, "D"),
            (400, "CD"),
            (100, "C"),
            (90, "XC"),
            (50, "L"),
            (40, "XL"),
            (10, "X"),
            (9, "IX"),
            (5, "V"),
            (4, "IV"),
            (1, "I")
        ]

    def intToRoman(self, num: int) -> str:
        res = ""
        for key, value in self.search:
            while num >= key:
                res += value
                num -= key
            if num == 0:
                break
        return res
# leetcode submit region end(Prohibit modification and deletion)
